trace/obfuscate.go

```diff
@@ -1,19 +1,61 @@
 package trace
 
 import (
+	"regexp"
+
 	pb "github.com/akitasoftware/akita-ir/go/api_spec"
 	"github.com/akitasoftware/akita-libs/spec_util"
 	. "github.com/akitasoftware/akita-libs/visitors"
 	vis "github.com/akitasoftware/akita-libs/visitors/http_rest"
 	"github.com/postmanlabs/postman-insights-agent/printer"
 )
 
-func obfuscate(m *pb.Method) {
-	var ov obfuscationVisitor
-	vis.Apply(&ov, m)
+type Obfuscator struct {
+	SensitiveDataKeys          []string
+	SensitiveDataValuePatterns []*regexp.Regexp
```

Comment:
Non-blocking: It looks like the jury is out on single-regexp vs. list-of-regexps., as the former can be more memory-efficient.  People have proposed some fairly bizarre solutions to avoid simple concatenation.

But, I would still like to explore combining the regular expressions so that we do a single search instead of dozens or hundreds of searches that restart each time. Maybe we can add that as a follow-up item if it's too difficult to work with at the moment.

Reply:
I looked for options for combining the regexp into a single one. We can do it by combining using `(|)` OR, but it won't be easy since we might have to add regex options, like case-insensitive, to some regexes, and I doubt that combining these regexes would be easy, CMIIW.
For now, I went forward with compiling the regexp and using it for comparing the string.